dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to Intel Corporation under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * Intel Corporation licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn
+
+import breeze.linalg.sum
+import com.intel.analytics.bigdl.nn.abstractnn.TensorModule
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+/**
+ * Reshape with the support of infered size, the infered dim is indicated as -1
+ * note that at most one infered dim is supported,
+ * the element number is not changed after reshape
+ * e.g. reshape from (3, 2) to (2, -1) is reshape to (2, 3)
```

Comment:
It's not easy to understand in the example. Please add more explanation to explain where is the 3 come from in the size after reshape.